### PR TITLE
io.jsonwebtoken.lang.UnknownClassException

### PR DESCRIPTION
**Pull request by RomanVolosiuk**
created: JwtTokenProvider
added dependency: jjwt-api

### CODE REVIEW

File: src/main/java/ua/volosiuk/mytokenservice/security/JwtTokenProvider.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.stream.Collectors;
+
+@Log4j2
+@Component
+public class JwtTokenProvider {
+    @Value("${jwt.token.secretKey}")
+    String secretKey;
+    private static final String SHA256 = "HmacSHA256";
+
+    public String createToken(String username, Role role){
```
Comment (mukonin) [inline src/main/java/ua/volosiuk/mytokenservice/security/JwtTokenProvider.java:27]: formatting

File: src/main/java/ua/volosiuk/mytokenservice/security/JwtTokenProvider.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        String headerAndPayload = getBase64UrlString(formatToString(header))
+                + "."
+                + getBase64UrlString(formatToString(claims));
+
+        return getBase64UrlString(formatToString(header))
+                + "."
+                + getBase64UrlString(formatToString(claims))
+                + "."
+                + sign(headerAndPayload)
+                ;
```
Comment (mukonin) [inline src/main/java/ua/volosiuk/mytokenservice/security/JwtTokenProvider.java:45]: here also formatting

File: src/main/java/ua/volosiuk/mytokenservice/security/JwtTokenProvider.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    private String formatToString(Map<String, String> map) {
```
Comment (mukonin) [inline src/main/java/ua/volosiuk/mytokenservice/security/JwtTokenProvider.java:48]: please use any of json mappers

File: src/main/java/ua/volosiuk/mytokenservice/security/JwtTokenProvider.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    private String getBase64UrlString(String originalString) {
+
+        return Base64.encodeBase64URLSafeString(originalString.getBytes(StandardCharsets.UTF_8));
+    }
+
+    private String sign(String data) {
+        try {
+            final Mac sha256HMAC = Mac.getInstance(SHA256);
```
Comment (mukonin) [inline src/main/java/ua/volosiuk/mytokenservice/security/JwtTokenProvider.java:63]: some of this vars could be class-level

File: pom.xml [hunk 5]
```diff
@@ -47,6 +47,17 @@
             <artifactId>json</artifactId>
             <version>20230227</version>
         </dependency>
+        <dependency>
+            <groupId>io.jsonwebtoken</groupId>
+            <artifactId>jjwt-api</artifactId>
+            <version>0.11.5</version>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.xml.bind</groupId>
```
Comment (mukonin) [inline pom.xml:56]: what is jakarta doing? where it is used?

Comment (RomanVolosiuk) [reply pom.xml:56]: It's imported by mistake

**Pull request merged**